Either/Either.swift

```diff
@@ -96,6 +96,38 @@ public func >>- <T, U, V> (either: Either<T, U>, @noescape transform: U -> Eithe
 }
 
 
+///  Extracts from a list of `Either` all the `Left` elements. All the `Left` elements are extracted in order.
+public func lefts <T, U>(list: [Either<T, U>]) -> [T] {
+	return list.reduce([T]()) { ( var memo, e) in
+		if let l = e.left {
+			memo.append(l)
+		}
+		
+		return memo
+	}
+}
+
+
+///  Extracts from a list of `Either` all the `Right` elements. All the `Right` elements are extracted in order.
+public func rights <T, U>(list: [Either<T, U>]) -> [U] {
+	return list.reduce([U]()) { ( var memo, e) in
+		if let l = e.right {
+			memo.append(l)
+		}
+		
+		return memo
+	}
+}
+
+
+///  Partitions a list of `Either` into two lists.
+///
+///  All the `Left` elements are extracted in order to the first component of the output. The `Right` elemests are extracted in order to the second component of the output.
+public func partitionEithers <T, U>(list: [Either<T, U>]) -> ([T], [U]) {
+	return (lefts(list), rights(list))
+}
```

Comment:
1. `static` functions are basically just namespaced functions. In these cases, all the parameter `Either`s have the same type as `Either` itself, which makes this appealing. You’d do something like:

	```swift
	extension Either {
		static func partition<S: SequenceType where S.Generator.Element == Either>(eithers: S) -> ([T], [U]) {
			return (lefts(eithers), rights(eithers))
		}
	}
	```

2. That makes sense to me :ok_hand:

3. The above appears to compile with `SequenceType`.

4. `lefts` and `rights` are essentially a specific combination of `filter` and `map`. Both of those are representable using `lazy`, and `SequenceOf` can wrap the result to erase the godawful types it would otherwise have:

	```swift
	return SequenceOf(lazy(eithers).filter { $0.left != nil }.map { $0.left! })
	```

	That aside, upon reflection I don’t think this sort of contortion is worth it. It’s valuable to produce a `CollectionType`, and `CollectionOf` doesn’t exist in stdlib (and is a pain in the ass to write).